pypesto/collections/collections.py

```diff
@@ -0,0 +1,484 @@
+            requested percentiles for a set of stacked simulations
+            """
+            summary = {}
+            summary[MEAN] = np.mean(tmp_array, axis=-1)
+            summary[STANDARD_DEVIATION] = np.std(tmp_array, axis=-1)
+            summary[MEDIAN] = np.median(tmp_array, axis=-1)
+            for perc in percentiles_list:
+                summary[f'{PERCENTILE} {perc}'] = np.percentile(tmp_array,
+                                                                perc, axis=-1)
+            return summary
+
+        # preallocate for results
+        cond_lists = {MEAN: [], STANDARD_DEVIATION: [], MEDIAN: []}
+        for perc in percentiles_list:
+            cond_lists[f'{PERCENTILE} {perc}'] = []
+
+        # iterate over conditions, compute summary
+        for i_cond in range(n_conditions):
+            # use some short hand
+            current_cond = self.prediction_results[0].conditions[i_cond]
+
+            # create a temporary array with all the outputs needed and wanted
+            tmp_output = _stack_outputs(i_cond)
+            tmp_output_sensi = _stack_outputs_sensi(i_cond)
+
+            # handle outputs
+            if tmp_output is not None:
+                output_summary = _compute_summary(tmp_output, percentiles_list)
+            else:
+                output_summary = {i_key: None for i_key in cond_lists.keys()}
+
+            # handle output sensitivities
+            if tmp_output_sensi is not None:
+                output_sensi_summary = _compute_summary(tmp_output_sensi,
+                                                        percentiles_list)
+            else:
+                output_sensi_summary = {i_key: None
+                                        for i_key in cond_lists.keys()}
+
+            # create some PredictionConditionResult to have an easier creation
+            # of PredictionResults for the summaries later on
+            for i_key in cond_lists.keys():
+                cond_lists[i_key].append(
+                    PredictionConditionResult(
+                        timepoints=current_cond.timepoints,
+                        output=output_summary[i_key],
+                        output_sensi=output_sensi_summary[i_key],
+                        observable_ids=current_cond.observable_ids
+                    )
+                )
+
+        self.prediction_summary = {i_key: PredictionResult(
+            conditions=cond_lists[i_key],
+            condition_ids=self.prediction_results[0].condition_ids,
+            comment=str(i_key))
+            for i_key in cond_lists.keys()
+        }
+
+        # also return the object
+        return self.prediction_summary
+
+
+class Collection:
+    """
+    A collection is of a thin wrapper around an numpy array.
+    """
+
+    def __init__(self,
+                 x_vectors: np.ndarray,
+                 x_names: Sequence[str] = None,
+                 vector_tags: Tuple[int, int] = None,
```

Comment:
```suggestion
                 vector_tags: Sequence[Tuple[int, int]] = None,
```

Should it be one tuple per vector?